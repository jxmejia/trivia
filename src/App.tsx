import { Global } from "@emotion/react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import { QuestionsContextProvider } from "./context";
import globalStyles from "./styles/global";
import { Home, Quiz, Results } from "./views";

function App(): JSX.Element {
  // TODO: Add error boundary https://www.npmjs.com/package/react-error-boundary
  return (
    <>
      <Global styles={globalStyles} />
      <QuestionsContextProvider>
        <BrowserRouter>
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/quiz" element={<Quiz />} />
            <Route path="/results" element={<Results />} />
          </Routes>
        </BrowserRouter>
      </QuestionsContextProvider>
    </>
  );
}

export default App;
