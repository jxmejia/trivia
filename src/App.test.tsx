import { render } from "@testing-library/react";
import App from "./App";

test("renders component", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
