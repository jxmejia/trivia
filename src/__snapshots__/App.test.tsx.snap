// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<div>
  <div
    class="css-1yltisf"
  >
    <div
      class="css-1ap9q9n"
    >
      <div
        class="css-1jbfuhw"
      >
        Welcome to the trivia challenge
      </div>
    </div>
    <div
      class="css-6hzeg0"
    >
      <div>
        <span
          class="css-mr3j8u"
        >
          You will be presented with 10 True or False questions.
        </span>
      </div>
      <div>
        <span
          class="css-mr3j8u"
        >
          Can you score 100%?
        </span>
      </div>
    </div>
    <div
      class="css-6hzeg0"
    >
      <button
        class="css-mr3j8u"
        type="button"
      >
        Begin
      </button>
    </div>
  </div>
</div>
`;
